src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -1648,19 +1648,25 @@ void SystemLayout::updateCrossBeams(System* system, LayoutContext& ctx)
         if (!mb->isMeasure()) {
             continue;
         }
+        bool somethingChanged = false;
```

Comment:
So if `somethingChanged` became `true` for one segment, it will still be `true` for all next segments in that measure. Is that intended, or should this variable be inside the `for(Segment& set : …)` loop to affect only the current segment and not the next ones?